src/imagecapturecore.cs

```diff
@@ -0,0 +1,901 @@
+//
+// Copyright 2018 Microsoft
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+//
+// imagecapturecore.cs: Bindings for the ImageCaptureCore API
+//
+using System;
+using AppKit;
+using Foundation;
+using ObjCRuntime;
+using CoreImage;
+using CoreGraphics;
+using CoreAnimation;
+
+namespace ImageCaptureCore {
+
+	interface IICDeviceDelegate {}
+
+	[Static]
+	interface ICDeviceCapabilities {
+		[Field ("ICDeviceCanEjectOrDisconnect")]
+		NSString DeviceCanEjectOrDisconnect { get; }
+
+		[Field ("ICCameraDeviceCanTakePicture")]
+		NSString CameraDeviceCanTakePicture { get; }
+
+		[Field ("ICCameraDeviceCanTakePictureUsingShutterReleaseOnCamera")]
+		NSString CameraDeviceCanTakePictureUsingShutterReleaseOnCamera { get; }
+
+		[Field ("ICCameraDeviceCanDeleteOneFile")]
+		NSString CameraDeviceCanDeleteOneFile { get; }
+
+		[Field ("ICCameraDeviceCanDeleteAllFiles")]
+		NSString CameraDeviceCanDeleteAllFiles { get; }
+
+		[Field ("ICCameraDeviceCanSyncClock")]
+		NSString CameraDeviceCanSyncClock { get; }
+
+		[Field ("ICCameraDeviceCanReceiveFile")]
+		NSString CameraDeviceCanReceiveFile { get; }
+
+		[Field ("ICCameraDeviceCanAcceptPTPCommands")]
+		NSString CameraDeviceCanAcceptPtpCommands { get; }
+	}
+
+	[Static]
+	interface ICScannerStatus {
+		[Field ("ICScannerStatusWarmingUp")]
+		NSString WarmingUp { get; }
+
+		[Field ("ICScannerStatusWarmUpDone")]
+		NSString WarmUpDone { get; }
+
+		[Field ("ICScannerStatusRequestsOverviewScan")]
+		NSString RequestsOverviewScan { get; }
+	}
```

Comment:
Same as above, smart enum?

Reply:
Would not be very useful since the way the API uses this is comparing keys from a protocol callback member, so having a smart enum would add an additional `.GetConstant ()` (`ICScannerStatus.WarmingUp.GetConstant ()`) that is not easily discoverable and unnecessary.

```csharp
[Export ("device:didReceiveStatusInformation:")]
public void DidReceiveStatusInformation (ICDevice device, NSDictionary<NSString, NSObject> status)
{
	var state = status[ICStatusNotificationKeys.NotificationKey] as NSString;
	if (state == ICScannerStatus.WarmingUp) { ... }
}
```